domains/auth/Auth.API/API/Repository/TokenStorage.cs

```diff
@@ -1,22 +1,32 @@
 using API.Models;
 using API.Repository;
 using FluentValidation;
+using Microsoft.Extensions.Caching.Memory;
 
 namespace API.TokenStorage;
 
 public class TokenStorage : ITokenStorage
 {
     private readonly IValidator<InternalToken> validator;
+    private readonly IMemoryCache memoryCache;
 
-    public TokenStorage(IValidator<InternalToken> validator)
+    public TokenStorage(IValidator<InternalToken> validator, IMemoryCache memoryCache)
     {
         this.validator = validator;
+        this.memoryCache = memoryCache;
     }
 
-    public void DeleteByOpaqueToken(string token)
-    {
-        throw new NotImplementedException();
-    }
+    public async Task<InternalToken> GetOrCreateInternalToken(InternalToken internalToken) =>
```

Comment:
Should this be deleted again as it seems that it is out of scope and not going to be needed?